package.json

```diff
@@ -15,7 +15,8 @@
     "postinstall": "lerna bootstrap && npx patch-package && ts-node scripts/link-ts-references.ts && npm run tsc && ts-node scripts/postinstall",
     "reinstall": "npm run clean && npm install",
     "start": "lerna exec --loglevel=silent --scope ganache -- npm run start --silent -- ",
-    "test": "lerna exec -- npm run test",
+    "test": "lerna exec --ignore @ganache/filecoin* -- npm run test",
+    "test.filecoin": "npx lerna exec --scope @ganache/filecoin* -- npm run test",
```

Comment:
Due to these requirements:
- Run all non-Filecoin tests on all NodeJS versions
- Only run Filecoin tests on NodeJS 12+ versions

That was realized where we can separate this into two scripts, hence these changes. Without these changes `npm run test` would fail on versions 12 and under.

If you would prefer that we don't use package.json scripts in the Github CI and manually call `lerna exec --ignore @ganache/filecoin* -- npm run test` instead of `npm run test` (and the same for filecoin), we can make those changes. I figured these changes were a more friendly dev experience (for me, others, grokability in the CI workflow documentation, etc) to clearly delineate the differences needed for CI to also be easily translatable in the NPM scripts. This allows developers, like yourself, that are focusing on supporting v10 to keep your node environment at v10 and still run `npm run test` and not have to bother with the filecoin changes.

Reply:
But now `npm test` doesn't run all tests that could affect filecoin, and I'd like for devs to be able to give a quick smoke test locally.

I'm going to chat internally about this a little more and get back to you!

Reply:
Ok, do note this decision was made with the `CONTRIBUTING.md` guidelines still telling devs to use `node@10.7.0` and that you were going to figure out whatever the base nodejs version because of this whole "minimum v12 for filecoin". It felt that requirement for v12+ was the oddball here, not running in a lesser environment. happy change it so we have `test` and `test.v10_11` 🤷. whatever you want!

Reply:
We could also make the `test` script more complicated to detect your nodejs version and change what gets ran

Reply:
Most of us don't like the idea of adding chain specific scripts to the root package.json, but no one came up with better alternatives.

My favorite idea is to make the filecoin package decide if its tests get run or not. Do you have any thoughts on a relatively "clean" and way of doing that? Or can you think of any big downsides to doing it this way?

Reply:
The cleanest way I could think of is yet another TS script that will check the current running NodeJS version; I couldn't find an environment variable for it, and not sure if checking an env variable in the command directly would even be windows friendly

The TS script could be a `is-atleast-node-12.ts` and has an exit code of 0 if it is and 1 otherwise and we can then use it like `ts-node is-atleast-node-12.ts && npm run test`. There could also be a whole dedicated TS script `test.ts` that checks the node version and then kicks off mocha as a sub process

Reply:
From the outside it should look like the tests passed when run via node 10 and 11, which might be the tricky part here.

(You could determine the node version via `node -e 'process.exit(parseInt(process.version.slice(1)) < 12 ? 1 : 0)'` (untested).

Note that I don't care if we stuff a bunch of npm scripts into filecoin's package.json. So if you could think of a cross-platform way of doing that, I think we can originate things so it's maintainable.

Reply:
> node -e 'process.exit(parseInt(process.version.slice(1)) < 12 ? 1 : 0)

This was the plan, but I was going to put it in a script because it is substantially cleaner than a long one-liner that hides in a json file. I can one-liner it if you'd prefer

Reply:
No, a file would be better, I think.